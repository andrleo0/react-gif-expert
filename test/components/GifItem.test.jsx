import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en el <GiftItem />', () => {

    const image = {
        title : 'Avatar',
        url : 'https://avatar/korra.jpg'
    }

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem {...image} />);
        expect( container ).toMatchSnapshot();
        
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render( <GifItem {...image} /> );
        const { src , alt } = screen.getByRole('img');
        expect( src ).toBe( image.url );
        expect( alt ).toBe( image.title );
        // expect( screen.getByRole('img').src ).toBe( image.url );
        // expect( screen.getByRole('img').alt ).toBe( image.title );

    })

    test('Debe de mostrar el titulo en el componente', () => { 
        render( <GifItem {...image} /> );
        expect( screen.getByText( image.title ) ).toBeTruthy();
    })

});