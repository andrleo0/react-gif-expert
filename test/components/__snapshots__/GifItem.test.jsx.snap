// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GiftItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Avatar"
      src="https://avatar/korra.jpg"
    />
    <p>
      Avatar
    </p>
  </div>
</div>
`;
